nautobot/extras/tests/test_plugins.py

```diff
@@ -466,6 +477,123 @@ def test_relationship_association_validator_raises_exception(self):
+        )
+
+        Site.objects.create(name="Site 1", slug="site-1", tenant=Tenant.objects.get(slug="tenant-1"))
+        Site.objects.create(name="Site 2", slug="site-2", tenant=Tenant.objects.get(slug="tenant-2"))
+        Site.objects.create(name="Site 3", slug="site-3", tenant=Tenant.objects.get(slug="tenant-3"))
+
+        Manufacturer.objects.create(name="Manufacturer 1", slug="manufacturer-1")
+        Manufacturer.objects.create(name="Manufacturer 2", slug="manufacturer-2")
+        Manufacturer.objects.create(name="Manufacturer 3", slug="manufacturer-3")
+
+        DeviceRole.objects.create(name="Device Role 1", slug="device-role-1")
+        DeviceRole.objects.create(name="Device Role 2", slug="device-role-2")
+        DeviceRole.objects.create(name="Device Role 3", slug="device-role-3")
+
+        DeviceType.objects.create(
+            manufacturer=Manufacturer.objects.get(slug="manufacturer-1"),
+            model="Model 1",
+            slug="model-1",
+            part_number="Part Number 1",
+            u_height=1,
+            is_full_depth=True,
+        )
+        DeviceType.objects.create(
+            manufacturer=Manufacturer.objects.get(slug="manufacturer-1"),
+            model="Model 2",
+            slug="model-2",
+            part_number="Part Number 2",
+            u_height=2,
+            is_full_depth=True,
+        )
+        DeviceType.objects.create(
+            manufacturer=Manufacturer.objects.get(slug="manufacturer-1"),
+            model="Model 3",
+            slug="model-3",
+            part_number="Part Number 3",
+            u_height=3,
+            is_full_depth=False,
+        )
+
+        Device.objects.create(
+            name="Device 1",
+            device_type=DeviceType.objects.get(slug="model-1"),
+            device_role=DeviceRole.objects.get(slug="device-role-1"),
+            tenant=Tenant.objects.get(slug="tenant-1"),
+            site=Site.objects.get(slug="site-1"),
+        )
+        Device.objects.create(
+            name="Device 2",
+            device_type=DeviceType.objects.get(slug="model-2"),
+            device_role=DeviceRole.objects.get(slug="device-role-2"),
+            tenant=Tenant.objects.get(slug="tenant-2"),
+            site=Site.objects.get(slug="site-2"),
+        )
+        Device.objects.create(
+            name="Device 3",
+            device_type=DeviceType.objects.get(slug="model-2"),
+            device_role=DeviceRole.objects.get(slug="device-role-3"),
+            tenant=Tenant.objects.get(slug="tenant-3"),
+            site=Site.objects.get(slug="site-3"),
+        )
+
+    def test_basic_custom_filter(self):
+        params = {"example_plugin_description": ["tenant-1.nautobot.com"]}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 1)
+
+    def test_added_lookup(self):
+        params = {"example_plugin_description__ic": ["tenant-1.nautobot"]}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 1)
```

Comment:
@glennmatthews does this satisfy the automagic generated lookup expressions? FilterSet field registered by the plugin is `example_plugin_description` and `*__ic` is created at register time.